server/src/main/java/org/elasticsearch/reservedstate/service/FileSettingsService.java

```diff
@@ -125,11 +125,18 @@ public Path watchedFile() {
      * <p>
      * If there's no file based settings file in this cluster, we'll remove all state reservations for
      * file based settings from the cluster state.
+     *
      * @param clusterState the cluster state before snapshot restore
-     * @param mdBuilder the current metadata builder for the new cluster state
-     * @param projectId the project associated with the restore
+     * @param builder      the current ClusterState builder for the new cluster state
+     * @param mdBuilder    the current metadata builder for the new cluster state
+     * @param projectId    the project associated with the restore
      */
-    public void handleSnapshotRestore(ClusterState clusterState, Metadata.Builder mdBuilder, ProjectId projectId) {
+    public void handleSnapshotRestore(
+        ClusterState clusterState,
+        ClusterState.Builder builder,
+        Metadata.Builder mdBuilder,
+        ProjectId projectId
+    ) {
```

Comment:
The parameters are getting a bit out of hand. They are all here to just support resetting the version of file settings. We can potentially extract an interface to represent them and pass different implementation based on whether the setup is multi-project. It does not need to be this PR. Could you please add a `@FixForMultiProject` annotation to it and we can work on that later.

Reply:
Could you please address this comment? Thanks!